packages/strapi-plugin-content-manager/admin/src/components/DynamicZone/index.js

```diff
@@ -78,6 +87,42 @@ const DynamicZone = ({
     [layout, name]
   );
 
+  const dynamicComponentCategories = useMemo(() => {
+    const categories = dynamicZoneAvailableComponents.reduce((cats, componentUid) => {
+      const {
+        category,
+        schema: { info },
+      } = getDynamicComponent(componentUid);
+
+      if (!cats[category]) {
+        cats[category] = [];
+      }
+      cats[category].push({
+        componentUid,
+        info,
+      });
+
+      return cats;
+    }, {});
+
+    return Object.keys(categories)
+      .sort()
+      .reduce((cats, cat) => {
+        cats.push({
+          category: cat,
+          components: categories[cat],
+        });
+
+        return cats;
+      }, []);
+  }, [dynamicZoneAvailableComponents, getDynamicComponent]);
+
+  useEffect(() => {
```

Comment:
With the modification added the effect is not needed here.

Reply:
@soupette done in 15570e056d4c6c19e4615bd214dd63fbff7df881

In the case there is only one category the list will not be opened by default.